eventually/authentication/models.py

```diff
@@ -0,0 +1,161 @@
+from django.db import models
+
+# Create your models here.
+
+from django.db import models
+from django.contrib.auth.base_user import AbstractBaseUser
+from django.utils.dateformat import format
+from  datetime import datetime
+
+class CustomUser(AbstractBaseUser):
+    """
+        This class represents a basic user, containing only the essential information about the user.\n
+        1. first, last and middle names: all Char with maximum length of 20 chars.
+        2. email: email field of max length 20 chars
+        3. password: charfield of max length 128 symbols
+        4. updated_at & created_at fields: DateTimeField.
+           created_at can't be changed once created
+    """
+
+    #id - already exists
+    first_name = models.CharField(('first name'), max_length = 20)
+    middle_name = models.CharField(('middle name'), max_length=20)
+    last_name = models.CharField(('second name'), max_length=20)
+    email = models.EmailField(('email address'), max_length = 40, unique = True)
+    password = models.CharField(('password'), max_length = 128)
+    updated_at = models.DateTimeField(auto_now_add = True)
+    created_at = models.DateTimeField(auto_now_add = True, editable = False)
+
+    def __str__(self):
+        """
+            This magic method is redefined to show first, last name and id of the CustomUser object.
+        """
+        return '{} {} {}'.format(self.first_name, self.last_name, self.id)
+
+    @staticmethod
+    def get_by_id(user_id):
+        """
+        :param user_id: the id of a user to be found in the DB
+        :return: user object or None if a user with such ID does not exist
+        """
+        try:
+            return CustomUser.objects.get(id=user_id)
+        except CustomUser.DoesNotExist:
+            return None
+
+    @staticmethod
+    def delete_by_id(user_id):
+        """
+        :param user_id: an id of a user to be deleted
+        :return: True if object existed in the db and was removed or False if it didn't exist
+        """
+        try:
+            user = CustomUser.get_by_id(user_id)
+            user.delete()
+            return True
+        except AttributeError:
+            return False
+
+    @staticmethod
+    def create(first_name, middle_name, last_name, email, password):
+        """
+        :param first_name: first name of a user
+        :param middle_name: middle name of a user
+        :param last_name: last name of a user
+        :param email: email of a user
+        :param password: password of a user
+        :return: a new user object which is also written into the DB
+        """
+        user = CustomUser(first_name=first_name,
+                          last_name=last_name,
+                          middle_name=middle_name,
+                          email=email,
+                          password=password)
+        user.save()
+        return user
+
+    #'FIXME: remove 4 methods below? Do we need them?'
```

Comment:
I think that four methods below aren't necessary because we can get all that info from `to_dict` method

Reply:
Makes sense. Methods removed.